src/hashtable.c

```diff
@@ -652,7 +652,7 @@ static int resize(hashtable *ht, size_t min_capacity, int *malloc_failed) {
     if (ht->type->rehashingStarted) ht->type->rehashingStarted(ht);
 
     /* If the old table was empty, the rehashing is completed immediately. */
-    if (ht->tables[0] == NULL || ht->used[0] == 0) {
+    if (ht->tables[0] == NULL) {
```

Comment:
If we want to keep the fast path to avoid iterating over the large table, we can add a check for `child_buckets[0] == 0` here. Is it worth it or is it a very unlikely case?

```suggestion
    if (ht->tables[0] == NULL || (ht->used[0] == 0 && ht->child_buckets[0] == 0)) {
```

Reply:
I think it depends. To trigger this scenario, you need to pause the shrink and remove entries from a hash table with at least 1 child bucket until there are 0 entries remaining. (e.g. expire dict delete remaining entries and pause the shrink in the meantime)  So if we reduce the shrink frequency accidentally, the probability of triggering may be higher. For now, I think this optimization can be added.